store/src/main/java/org/apache/rocketmq/store/MappedFileQueue.java

```diff
@@ -144,35 +145,39 @@ void deleteExpiredFile(List<MappedFile> files) {
         }
     }
 
+
     public boolean load() {
         File dir = new File(this.storePath);
-        File[] files = dir.listFiles();
-        if (files != null) {
-            // ascending order
-            Arrays.sort(files);
-            for (File file : files) {
-
-                if (file.length() != this.mappedFileSize) {
-                    log.warn(file + "\t" + file.length()
-                        + " length not matched message store config value, please check it manually");
-                    return false;
-                }
+        File[] ls = dir.listFiles();
+        if (ls != null) {
+            return doLoad(Arrays.asList(ls));
+        }
+        return true;
+    }
 
-                try {
-                    MappedFile mappedFile = new MappedFile(file.getPath(), mappedFileSize);
-
-                    mappedFile.setWrotePosition(this.mappedFileSize);
-                    mappedFile.setFlushedPosition(this.mappedFileSize);
-                    mappedFile.setCommittedPosition(this.mappedFileSize);
-                    this.mappedFiles.add(mappedFile);
-                    log.info("load " + file.getPath() + " OK");
-                } catch (IOException e) {
-                    log.error("load file " + file + " error", e);
-                    return false;
-                }
+    public boolean doLoad(List<File> files) {
+        // ascending order
+        Collections.sort(files);
```

Comment:
According to the code :
![image](https://user-images.githubusercontent.com/14816818/134317644-4061cb3f-5470-43bf-bf52-ee974458db10.png)
if paths are a/, b/ then files will be created such as:
a/000000   b/000001  a/000002  b/000003
so sort these can cause wrong order.

Reply:
Great point.
Changed `Collections.sort(files)` to `files.sort(Comparator.comparing(File::getName))`,  sort the file using simple filenames, aka, the last part of the path.

And added test in org.apache.rocketmq.store.MultiPathMappedFileQueueTest#testLoadReadOnlyMappedFiles， PTAL.